packages/autocomplete-plugin-redirect-url/src/createRedirectUrlPlugin.ts

```diff
@@ -0,0 +1,159 @@
+import {
+  BaseItem,
+  OnSelectParams,
+  OnSubmitParams,
+} from '@algolia/autocomplete-core';
+import { AutocompletePlugin } from '@algolia/autocomplete-js';
+import { warn } from '@algolia/autocomplete-shared';
+
+import { defaultTemplates } from './templates';
+import {
+  CreateRedirectUrlPluginParams,
+  OnRedirectOptions,
+  RedirectUrlItem,
+  RedirectUrlPlugin as RedirectUrlPluginData,
+  TransformResponseParams,
+} from './types';
+
+function defaultTransformResponse<THit>(
+  response: TransformResponseParams<THit>
+): string | undefined {
+  return (response as Record<string, any>).renderingContent?.redirect?.url;
+}
+
+function defaultOnRedirect(
+  redirects: RedirectUrlItem[],
+  { navigator, state }: OnRedirectOptions<RedirectUrlItem>
+) {
+  const itemUrl = redirects[0]?.urls?.[0];
+  if (itemUrl) {
+    navigator.navigate({ itemUrl, item: redirects[0], state });
+  }
+}
```

Comment:
We need to take into account modifier keys to remain usable with the navigator. This way you're able to open link in new tabs and new windows with keyboard modifiers.

```suggestion
function defaultOnRedirect(
  redirects: RedirectUrlItem[],
  { navigator, state, event }: OnRedirectOptions<RedirectUrlItem>
) {
  const item = redirects[0];
  const itemUrl = item?.urls?.[0];

  if (!itemUrl) {
    return;
  }

  if (event.metaKey || event.ctrlKey) {
    navigator.navigateNewTab({ itemUrl, item, state });
  } else if (event.shiftKey) {
    navigator.navigateNewWindow({ itemUrl, item, state });
  } else {
    navigator.navigate({ itemUrl, item, state });
  }
}
```

Reply:
(You'll need to pass the `event` to the calls now.)

Reply:
Yeah, I learned that the had way... Should be good now.

Reply:
Maybe we should move that into the navigate function itself in the future?